src/xrGame/Weapon.cpp

```diff
@@ -1401,6 +1408,9 @@ void CWeapon::UpdateAddonsVisibility()
     }
 
     bone_id = pWeaponVisual->LL_BoneID(wpn_grenade_launcher);
+    if (bone_id == BI_NONE && ShadowOfChernobylMode)
+        wpn_grenade_launcher = WPN_GRENADE_LAUNCHER_SOC;
```

Comment:
The changes here should be the same as in the comment for `UpdateHUDAddonsVisibility` function.